doc/sphinx/source/mlir.rst

```diff
@@ -0,0 +1,79 @@
+.. _mlir-label:
+
+=========================
+Experimental MLIR Support
+=========================
+
+There is highly experimental support for LLVM's MLIR project.
+The motivation for this work is to use MLIR's parallel dialects,
+which enables us to plug into existing MLIR optimization and backend frameworks.
+Another motivation is the availability of MLIR-based high-level synthesis tools such as
+`CIRCT <https://github.com/llvm/circt>`_ or `HIDA <https://github.com/UIUC-ChenLab/ScaleHLS-HIDA>`_.
+
+The MLIR-based flow uses a different front-end compiler than PoCL's LLVM-based flow.
+There are two supported front-end options: Polygeist, and ClangIR
+(with experimental *ThroughMLIR*-compilation).
+The front-end produces SPMD programs that use upstream MLIR dialects such as affine, arith, and scf.
+
+The MLIR-based compiler links in built-in functions pre-compiled to MLIR IR from `lib/kernel/mlir`.
+Then it performs a de-SPMD conversion using a pass in `lib/mliropencl/lib/Transforms/Workgroup.cpp`.
+After that, any barrier operations are eliminated using
+`a pass ported from Polygeist <https://github.com/cpc/clangir/blob/main/mlir/lib/Polygeist/Transforms/DistributeBarriers.cpp>`_
+
+Then, the MLIR IR is lowered to a specific device.
+For CPU devices it is lowered to LLVM IR compatible with the PoCL's typical LLVM-based flow.
+After that, it is executed on the CPU device using the same codepath as the LLVM-based flow.
+
+Building
+^^^^^^^^
+
+The current version requires `the specific ClangIR-branch <https://github.com/cpc/clangir>`_.
+
+When building ClangIR, set::
+
+  -DLLVM_ENABLE_PROJECTS="mlir;clang"
+  -DCLANG_ENABLE_CIR=ON
+  -DLLVM_ENABLE_RTTI=ON
```

Comment:
Is RTTI required to get MLIR+CIR? It seems to me [it is not required](https://llvm.github.io/clangir/GettingStarted/build-install.html).

Reply:
You are right. I needed to add it earlier to get some symbols properly exported, but re-testing it now, it seems to work without it as well. Not sure what exactly I have changed since. Removed the requirement.